### PR TITLE
Sort clause format change

### LINKED ISSUE #323: Change sort clause format
Implement sort clause to be json document { "active_user" : -1 } where 1 represent ascending and -1 represent descending order.

### PR DESCRIPTION
**Pull request by maheshrajamani**
**What this PR does**:
Sort clause format changed from array to object.
**Which issue(s) this PR fixes**:
Fixes #323 

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: src/main/java/io/stargate/sgv2/jsonapi/api/model/command/clause/sort/SortClause.java [hunk 1]
```diff
@@ -14,9 +14,9 @@
  */
 @JsonDeserialize(using = SortClauseDeserializer.class)
 @Schema(
-    type = SchemaType.ARRAY,
-    implementation = String[].class,
+    type = SchemaType.OBJECT,
+    implementation = Object.class,
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/api/model/command/clause/sort/SortClause.java:18]: Should this be `java.util.Map`? (JSON Object -> Java Map)

**Review by tatu-at-datastax — APPROVED** [review]
LGTM

File: src/main/java/io/stargate/sgv2/jsonapi/api/model/command/clause/sort/SortClause.java (hunk 1 above)
Comment (maheshrajamani) [reply src/main/java/io/stargate/sgv2/jsonapi/api/model/command/clause/sort/SortClause.java:18]: OK will change, think need to fix it in FilterClause too.

File: src/main/java/io/stargate/sgv2/jsonapi/api/model/command/deserializers/SortClauseDeserializer.java [hunk 2]
```diff
@@ -35,36 +38,37 @@ public SortClause deserialize(JsonParser parser, DeserializationContext ctxt)
...
-    List<SortExpression> expressions = new ArrayList<>();
-    for (JsonNode inner : node) {
-      if (!inner.isTextual()) {
+    Iterator<Map.Entry<String, JsonNode>> fieldIter = sortNode.fields();
+    List<SortExpression> sortExpressions = new ArrayList<>(sortNode.size());
+    while (fieldIter.hasNext()) {
+      Map.Entry<String, JsonNode> inner = fieldIter.next();
+
+      if (!inner.getValue().isInt()
+          || !(inner.getValue().asInt() == 1 || inner.getValue().asInt() == -1)) {
```
[the remark this answers is not in the document]
Comment (maheshrajamani) [reply src/main/java/io/stargate/sgv2/jsonapi/api/model/command/deserializers/SortClauseDeserializer.java:55]: Understood. Will change it.

**Review by vkarpov15 — COMMENTED** [review]
LGTM :+1:

**Pull request merged**